crates/router/src/core/mandate.rs

```diff
@@ -243,7 +226,80 @@ where
         _ => Some(router_data.request.get_payment_method_data()),
     }
 }
+pub async fn update_mandate_procedure<F, FData>(
+    state: &AppState,
+    resp: types::RouterData<F, FData, types::PaymentsResponseData>,
+    mandate: Mandate,
+    merchant_id: &str,
+    pm_id: Option<String>,
+) -> errors::RouterResult<types::RouterData<F, FData, types::PaymentsResponseData>>
+where
+    FData: MandateBehaviour,
+{
+    let mandate_details = match resp.response.clone() {
```

Comment:
Don't clone the entire struct. Match on a reference, and then clone the specific required data (here, mandate_reference)
```suggestion
    let mandate_details = match &resp.response {
```